src/docfx/lib/github/GitHubAccessor.cs

```diff
@@ -58,87 +56,33 @@ public async Task<(Error, GitHubUser)> GetUserByLogin(string login)
-                            email = "",
-                            login = "",
-                            databaseId = 0,
-                        },
-                    },
-                });
-
-                if (grahpApiResponse.errors != null && grahpApiResponse.errors.Length != 0)
-                {
-                    var notFoundError = grahpApiResponse.errors.FirstOrDefault(e => e.type == "NOT_FOUND" && e.path.Contains("user"));
-                    if (notFoundError != null)
-                        return default;
-
-                    var rateLimitError = grahpApiResponse.errors.FirstOrDefault(e => e.type == "MAX_NODE_LIMIT_EXCEEDED" || e.type == "RATE_LIMITED");
-                    if (rateLimitError != null)
-                    {
-                        _rateLimitError = Errors.GitHubApiFailed(rateLimitError.message);
-                    }
+            var (error, data) = await Query(
+                query,
+                new { login },
+                new { user = new { name = "", email = "", login = "", databaseId = 0 } });
 
-                    return (Errors.GitHubApiFailed(rateLimitError?.message ?? grahpApiResponse.errors.First().message), null);
-                }
-
-                return (null, new GitHubUser
-                {
-                    Id = grahpApiResponse.data.user.databaseId,
-                    Login = grahpApiResponse.data.user.login,
-                    Name = string.IsNullOrEmpty(grahpApiResponse.data.user.name) ? grahpApiResponse.data.user.login : grahpApiResponse.data.user.name,
-                    Emails = !string.IsNullOrEmpty(grahpApiResponse.data.user.email) ? new[] { grahpApiResponse.data.user.email } : Array.Empty<string>(),
-                });
-            }
-            catch (Exception ex)
+            if (error != null || data?.user is null)
             {
-                Log.Write(ex);
-                return (Errors.GitHubApiFailed(ex.InnerException?.Message ?? ex.Message), null);
+                return (error, null);
             }
-        }
 
-        public async Task<(Error, IEnumerable<GitHubUser>)> GetUsersByCommit(string owner, string name, string commit)
-        {
-            if (_rateLimitError != null)
+            return (null, new GitHubUser
             {
-                return default;
-            }
+                Id = data.user.databaseId,
+                Login = data.user.login,
+                Name = string.IsNullOrEmpty(data.user.name) ? data.user.login : data.user.name,
+                Emails = new[] { data.user.email }.Where(email => !string.IsNullOrEmpty(email)).ToArray(),
+            });
+        }
 
-            if (_unauthorizedError != null)
+        public async Task<(Error, IEnumerable<GitHubUser>)> GetUsersByCommit(string owner, string name, string commit, string authorEmail = null)
+        {
+            if (_fatalError != null)
             {
-                return default;
+                return (_fatalError, default);
```

Comment:
>_fatalError [](start = 24, length = 11)

that will cause too many same errors reported

Reply:
Same as before, `github-api-failed` is a warning, @superyyrrzz originally wants to let users know which articles are affected by github failure by report the error for each affected article.

Reply:
hmm, seems we've updated the behavior to only report the first error.